sys/net/network_layer/ng_ipv6/netif/ng_ipv6_netif.c

```diff
@@ -474,14 +475,24 @@ void ng_ipv6_netif_init_by_dev(void)
                            sizeof(if_type)) != -ENOTSUP) &&
             (if_type == NG_NETTYPE_SIXLOWPAN)) {
             uint16_t src_len = 8;
-            DEBUG("Set 6LoWPAN flag\n");
+            uint16_t max_frag_size = UINT16_MAX;
+
+            DEBUG("ipv6 netif: Set 6LoWPAN flag\n");
             ipv6_ifs[i].flags |= NG_IPV6_NETIF_FLAGS_SIXLOWPAN;
             /* use EUI-64 (8-byte address) for IID generation and for sending
              * packets */
             ng_netapi_set(ifs[i], NETCONF_OPT_SRC_LEN, 0, &src_len,
                           sizeof(src_len)); /* don't care for result */
-        }
 
+            if (ng_netapi_get(ifs[i], NETCONF_OPT_MAX_PACKET_SIZE,
+                              0, &max_frag_size, sizeof(max_frag_size)) < 0) {
+                /* if error we assume it works */
```

Comment:
> How does it do so?

Either by being set by the user (though there is arguably no command for this yet) or by getting it from the device. If both are not the case the default value is 1280.

> If this value is only used for 6LoWPAN fragmentation, I would vote for either exiting here or setting this value to a fixed size.

I do the later here ;-). But seriously, after this discussion and reviewing what happens to a packet as it traverses the stack I do think `UINT16_MAX` is the most sensible solution. Say the link layer has no maximum packet length because it supports streaming. Then UINT16_MAX as the size of a single maxed out IPv6 packet (even that is not true because the header is missing from that the) Is the maximum fragment size 6LoWPAN can send. If not it either can tell both IPv6 and 6LoWPAN this through this option or it is always 1280 (as you proposed earlier) which all will fail the "should I fragment" test because they are all lesser than UINT16_MAX.